Feature/static site

jnanic opened the pull request:
Only until 20 Nov. config files enroute,

jnanic committed:
Initial commit with just run.py

jnanic committed:
Static files for the static site. Only until 20 Nov

jnanic committed:
deleted old files

bogderp commented:
Won't get to this until wednesday night but just looking at it.

1. remove the testflasksite.tk directory. 
2. clean up your js and css. if you are using webpack to generate your js/css include any necessary files. 
3. don't put dynamic js/css under version control, generate them on your dev/prod server when deploying your site. 
4. upload your css/js assets that are being compiled into your compressed dynamic js/css
5. write a brief readme on how to generate dynamic assets

@HarrisonSanders @yash1337

jnanic committed:
new files with new folder name

jnanic committed:
added precompiled files and changed folder stucture

HavenTheWanderer committed:
In theory, this should be what is needed to make the app run. Hopefully, the .gitignore is working.

bogderp commented on README.md:
```diff
@@ -1 +1,44 @@
+
+**Once set up, only Postgres.app and the rails server need to be on to run the site**
+    
+    rails server
```
Do you really need this?

nathankurt replied:
I mean I think the http://localhost:4200 thing is helpful right? isn't it normally localhost:5000?

bogderp replied:
I'm referring to rails. I don't think angular has a rails dependency

bogderp replied:
for reference, i place my comments on the offending line

HavenTheWanderer replied:
Yeah, I copy-pasted an old readme and missed that, sorry!

bogderp commented on package.json:
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "sparta-hack",
```
nit: sparta-hack makes me cringe. this should be spartahack

bogderp commented on src/app/app.component.html:
```diff
@@ -0,0 +1,10 @@
+</div>
+<app-top></app-top>
+<div [@fadeIn]=''>
+    <app-faq></app-faq>
```
just a comment, do you need to have all your components start with `app-`. could you not rather have the prefix be page specific or omit it all together?

HavenTheWanderer replied:
It's an Angular convention. As an app gets more complex I usually abandon it, but I use it to denote the stuff that is on the very top

bogderp commented on src/app/header/header.component.html:
```diff
@@ -0,0 +1,27 @@
+		</div>
+		<div class="medium-2 small-3 columns menu-button">
+			<a scrollTo href="#faq">
+				FAQ's
```
faq stands for frequently asked questions so the `'s` is unnecessary

bogderp commented on src/app/services/faq.service.ts:
```diff
@@ -0,0 +1,42 @@
+
+  getFaqs(): Observable<Faq[]> {
+    let options = new RequestOptions({ headers : this.headers});
+    return this.http.get('https://api.elephant.spartahack.com/faqs', options)
```
you should probably take this from an environment file or config. Your production server will not be this endpoint

bogderp commented on src/app/services/sponsor.service.ts:
```diff
@@ -0,0 +1,36 @@
+
+  headers = new Headers({
+    // 'content-type': 'application/json',
+    // 'authorization': 'Token token=9ba5cc744d40feb58d0bc6bce7c8d8be',
```
also this token should never be publicly available. This is the dev token so it's fine. but the production token should not be accessible by the public

HavenTheWanderer replied:
The build process takes comments out of the typescript, but that's good to know!

bogderp commented on src/app/top/top.component.html:
```diff
@@ -0,0 +1,53 @@
+<div class="row">
+    <section id="mail">
+    	<!-- MailChimp Sign Up -->
+        <div id="mc_embed_signup" class="small-8 small-offset-2 subscriber" (click)="OnSignUpClick()" [@mailButtonIn]="">
```
i've noticed on the testflasksite this portion is unstyled.

bogderp commented on src/assets/data/faqs.json:
```diff
@@ -0,0 +1,222 @@
+﻿[
```
if you are using the api to get faqs why do you need this

HavenTheWanderer replied:
We don't.

bogderp commented:
ahh, yash told me otherwise

> On Nov 9, 2017, at 9:49 PM, HarrisonSanders <notifications@github.com> wrote:
> 
> @HarrisonSanders commented on this pull request.
> 
> In src/assets/data/faqs.json <https://github.com/SpartaHack/SpartaHackIV-Website/pull/2#discussion_r150143671>:
> 
> > @@ -0,0 +1,222 @@
> +﻿[
> We don't.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/SpartaHack/SpartaHackIV-Website/pull/2#discussion_r150143671>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AI6Nuu59JhTJyc7HG_dRvBvxVQkh4QvZks5s07mugaJpZM4QUQXq>.
>

bogderp commented on src/index.html:
```diff
@@ -0,0 +1,14 @@
+  <meta charset="utf-8">
+  <title>SpartaHack IV</title>
+  <base href="/">
+  <link href="http://cdn-images.mailchimp.com/embedcode/classic-10_7.css" rel="stylesheet" type="text/css">  
```
i suggest removing this and styling the form yourself

bogderp reviewed (dismissed):
it's a solid intro with the logo and snowflakes fading in but everything after that is lacking.

Need more spacing between date and registration.
mailing list needs styling.
faq needs to be styled
  I can't tell what question corresponds to the answer

white is blinding, black was nicer imo, you could see the snowflakes falling more easily.

transparent buttons at the bottom should match the mailing list button. 

mobile styling is super awkward on an iphone 5.

HavenTheWanderer committed:
Minor updates, as per Bogdans requests. Also, added a really important hidden file that actually makes the darn thing run.
